src/GraphQL/Violation/ViolationCollectionInterface.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\graphql\GraphQL\Violation;
+
+/**
+ * A collection of violations that can be triggered by a GraphQL mutation.
+ */
+interface ViolationCollectionInterface {
```

Comment:
should we combine this with ResponseInterface? Then we don't need this extra class. A data producer can just add violations on the mutation response object itself.

Reply:
moved, looks better now